enhancements/network/multi-network-vrfs.md

```diff
@@ -0,0 +1,924 @@
+a pod connected to the default cluster network (100.64.0.0/16), a user-defined primary network (100.65.0.0/16), and a
+secondary network (100.66.0.0/16), the routing table would look like:
+
+```
+[root@pod3 /]# ip route show
+default via 10.245.1.1 dev eth1 
+10.96.0.0/16 via 10.245.1.1 dev eth1
+10.244.0.0/16 via 10.244.1.1 dev eth0
+10.245.0.0/16 via 10.245.1.1 dev eth1
+10.246.0.0/16 via 10.246.1.1 dev eth2
+10.244.0.0/24 dev eth0 proto kernel scope link src 10.245.0.8
+10.245.0.0/24 dev eth1 proto kernel scope link src 10.245.0.8
+10.246.0.0/24 dev eth1 proto kernel scope link src 10.245.0.8
+100.64.0.0/16 via 10.244.1.1 dev eth0
+100.65.0.0/16 via 10.244.2.1 dev eth1
+100.66.0.0/16 via 10.244.3.1 dev eth2
+```
+
+Therefore when specifying a user-defined network it will be imperative to ensure that the networks a pod will connect to
+do not have overlapping pod network or join network subnets. OVN-Kubernetes should be able to detect this scenario and
+refuse to CNI ADD a pod with conflicts.
+
+### DNS
+
+DNS lookups will happen via every pod’s access to the DNS service on the cluster default network. CoreDNS lookups for
+pods will resolve to the pod’s IP on the cluster default network. This is a limitation of the first phase of this feature
+and will be addressed in a future enhancement. DNS lookups for services and external entities will function correctly.
+
+### Services
+
+Services in Kubernetes are namespace scoped. Any creation of a service in a namespace without a user-defined network
+(using cluster default network as primary) will only be accessible by other namespaces also using the default network as
+their primary network. Services created in namespaces served by user-defined networks, will only be accessible to
+namespaces connected to the user-defined network.
+
+Since most applications require DNS and KAPI access, there is an exception to the above conditions where pods that are
+connected to user-defined networks are still able to access KAPI and DNS services that reside on the cluster default
+network. In the future, access to more services on the default network may be granted. However, that would require more
+groundwork around enforcing network policy (which is evaluated typically after service DNAT) as potentially nftables
+rules. Such work is considered a future enhancement and beyond the scope of this initial implementation.
+
+With this proposal, OVN-Kubernetes will check which network is being used for this namespace, and then only enable the
+service there. The cluster IP of the service will only be available in the network of that service. Host networked pods
+in a namespace with a user-defined primary network will also be limited to only accessing the cluster IP of the services
+for that network it. Load balancer IP and nodeport services are also supported on user-defined networks. Service
+selectors are only able select endpoints from the same namespace where the service exists. Services that exist before
```

Comment:
this will still work, as long as the endpoint is reachable outside the cluster or on the user-defined network. It will not work for endpoints in other networks. I used to sometimes test creating my own endpoints (before endpoint slices) and it was broken in OVNK. I'm not sure we even test it. I dont think we need to worry about it for now.